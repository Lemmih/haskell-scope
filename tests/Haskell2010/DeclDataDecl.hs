{-# LANGUAGE Haskell2010 #-}
module DeclDataDecl where

class Show a

data T1
data T2 a
data T3 a = T3 a
data T4 deriving Show
data T5 = T5 deriving Show

newtype T6 = T6 T1
newtype T7 a = T7 (T2 a)
newtype T8 = T8 T1 deriving Show

data T9 a b = Left a | Right b

data T10 a b = T10 { field1 :: a, field2 :: b}
data Show a => T11 a = T11 a

-- Not well-typed. Just used to verify name resolution.
collection :: (T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11)
collection = [T3, T5, T6, T7, T8, Left, Right, T10, field1, field2, T11]
