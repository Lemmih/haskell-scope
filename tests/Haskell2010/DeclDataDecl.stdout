Scope errors:

Definitions:
  Definition 1, from DeclDataDecl.decl0:
    class [4mShow[0m a
  Definition 2, from DeclDataDecl.decl0:
    class Show [4ma[0m
  Definition 3, from DeclDataDecl.decl1:
    data [4mT1[0m
  Definition 4, from DeclDataDecl.decl2:
    data [4mT2[0m a
  Definition 5, from DeclDataDecl.decl2:
    data T2 [4ma[0m
  Definition 6, from DeclDataDecl.decl3:
    data [4mT3[0m a = T3 a
  Definition 7, from DeclDataDecl.decl3:
    data T3 [4ma[0m = T3 a
  Definition 8, from DeclDataDecl.decl3:
    data T3 a = [4mT3[0m a
  Definition 9, from DeclDataDecl.decl4:
    data [4mT4[0m deriving Show
  Definition 10, from DeclDataDecl.decl5:
    data [4mT5[0m = T5 deriving Show
  Definition 11, from DeclDataDecl.decl5:
    data T5 = [4mT5[0m deriving Show
  Definition 12, from DeclDataDecl.decl6:
    newtype [4mT6[0m = T6 T1
  Definition 13, from DeclDataDecl.decl6:
    newtype T6 = [4mT6[0m T1
  Definition 14, from DeclDataDecl.decl7:
    newtype [4mT7[0m a = T7 (T2 a)
  Definition 15, from DeclDataDecl.decl7:
    newtype T7 [4ma[0m = T7 (T2 a)
  Definition 16, from DeclDataDecl.decl7:
    newtype T7 a = [4mT7[0m (T2 a)
  Definition 17, from DeclDataDecl.decl8:
    newtype [4mT8[0m = T8 T1 deriving Show
  Definition 18, from DeclDataDecl.decl8:
    newtype T8 = [4mT8[0m T1 deriving Show
  Definition 19, from DeclDataDecl.decl9:
    data [4mT9[0m a b = Left a | Right b
  Definition 20, from DeclDataDecl.decl9:
    data T9 [4ma[0m b = Left a | Right b
  Definition 21, from DeclDataDecl.decl9:
    data T9 a [4mb[0m = Left a | Right b
  Definition 22, from DeclDataDecl.decl9:
    data T9 a b = [4mLeft[0m a | Right b
  Definition 23, from DeclDataDecl.decl9:
    data T9 a b = Left a | [4mRight[0m b
  Definition 24, from DeclDataDecl.decl10:
    data [4mT10[0m a b = T10 { field1 :: a, field2 :: b}
  Definition 25, from DeclDataDecl.decl10:
    data T10 [4ma[0m b = T10 { field1 :: a, field2 :: b}
  Definition 26, from DeclDataDecl.decl10:
    data T10 a [4mb[0m = T10 { field1 :: a, field2 :: b}
  Definition 27, from DeclDataDecl.decl10:
    data T10 a b = [4mT10[0m { field1 :: a, field2 :: b}
  Definition 28, from DeclDataDecl.decl10:
    data T10 a b = T10 { [4mfield1[0m :: a, field2 :: b}
  Definition 29, from DeclDataDecl.decl10:
    data T10 a b = T10 { field1 :: a, [4mfield2[0m :: b}
  Definition 30, from DeclDataDecl.decl11:
    data Show a => [4mT11[0m a = T11 a
  Definition 31, from DeclDataDecl.decl11:
    data Show a => T11 [4ma[0m = T11 a
  Definition 32, from DeclDataDecl.decl11:
    data Show a => T11 a = [4mT11[0m a
  Definition 33, from DeclDataDecl.decl13:
    [4mcollection[0m = [T3, T5, T6, T7, T8, Left, Right, T10, field1, field2, T11]

Use sites:
  Definition used: 7
    data T3 a = T3 [4ma[0m
  Definition used: 1
    data T4 deriving [4mShow[0m
  Definition used: 1
    data T5 = T5 deriving [4mShow[0m
  Definition used: 3
    newtype T6 = T6 [4mT1[0m
  Definition used: 4
    newtype T7 a = T7 ([4mT2[0m a)
  Definition used: 15
    newtype T7 a = T7 (T2 [4ma[0m)
  Definition used: 3
    newtype T8 = T8 [4mT1[0m deriving Show
  Definition used: 1
    newtype T8 = T8 T1 deriving [4mShow[0m
  Definition used: 20
    data T9 a b = Left [4ma[0m | Right b
  Definition used: 21
    data T9 a b = Left a | Right [4mb[0m
  Definition used: 25
    data T10 a b = T10 { field1 :: [4ma[0m, field2 :: b}
  Definition used: 26
    data T10 a b = T10 { field1 :: a, field2 :: [4mb[0m}
  Definition used: 1
    data [4mShow[0m a => T11 a = T11 a
  Definition used: 31
    data Show [4ma[0m => T11 a = T11 a
  Definition used: 31
    data Show a => T11 a = T11 [4ma[0m
  Definition used: 33
    [4mcollection[0m :: (T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11)
  Definition used: 3
    collection :: ([4mT1[0m, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11)
  Definition used: 4
    collection :: (T1, [4mT2[0m, T3, T4, T5, T6, T7, T8, T9, T10, T11)
  Definition used: 6
    collection :: (T1, T2, [4mT3[0m, T4, T5, T6, T7, T8, T9, T10, T11)
  Definition used: 9
    collection :: (T1, T2, T3, [4mT4[0m, T5, T6, T7, T8, T9, T10, T11)
  Definition used: 10
    collection :: (T1, T2, T3, T4, [4mT5[0m, T6, T7, T8, T9, T10, T11)
  Definition used: 12
    collection :: (T1, T2, T3, T4, T5, [4mT6[0m, T7, T8, T9, T10, T11)
  Definition used: 14
    collection :: (T1, T2, T3, T4, T5, T6, [4mT7[0m, T8, T9, T10, T11)
  Definition used: 17
    collection :: (T1, T2, T3, T4, T5, T6, T7, [4mT8[0m, T9, T10, T11)
  Definition used: 19
    collection :: (T1, T2, T3, T4, T5, T6, T7, T8, [4mT9[0m, T10, T11)
  Definition used: 24
    collection :: (T1, T2, T3, T4, T5, T6, T7, T8, T9, [4mT10[0m, T11)
  Definition used: 30
    collection :: (T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, [4mT11[0m)
  Definition used: 8
    collection = [[4mT3[0m, T5, T6, T7, T8, Left, Right, T10, field1, field2, T11]
  Definition used: 11
    collection = [T3, [4mT5[0m, T6, T7, T8, Left, Right, T10, field1, field2, T11]
  Definition used: 13
    collection = [T3, T5, [4mT6[0m, T7, T8, Left, Right, T10, field1, field2, T11]
  Definition used: 16
    collection = [T3, T5, T6, [4mT7[0m, T8, Left, Right, T10, field1, field2, T11]
  Definition used: 18
    collection = [T3, T5, T6, T7, [4mT8[0m, Left, Right, T10, field1, field2, T11]
  Definition used: 22
    collection = [T3, T5, T6, T7, T8, [4mLeft[0m, Right, T10, field1, field2, T11]
  Definition used: 23
    collection = [T3, T5, T6, T7, T8, Left, [4mRight[0m, T10, field1, field2, T11]
  Definition used: 27
    collection = [T3, T5, T6, T7, T8, Left, Right, [4mT10[0m, field1, field2, T11]
  Definition used: 28
    collection = [T3, T5, T6, T7, T8, Left, Right, T10, [4mfield1[0m, field2, T11]
  Definition used: 29
    collection = [T3, T5, T6, T7, T8, Left, Right, T10, field1, [4mfield2[0m, T11]
  Definition used: 32
    collection = [T3, T5, T6, T7, T8, Left, Right, T10, field1, field2, [4mT11[0m]

Exports:
  Show, Class from DeclDataDecl
  T1, Data from DeclDataDecl
  T10, Data from DeclDataDecl
  T11, Data from DeclDataDecl
  T2, Data from DeclDataDecl
  T3, Data from DeclDataDecl
  T4, Data from DeclDataDecl
  T5, Data from DeclDataDecl
  T6, Data from DeclDataDecl
  T7, Data from DeclDataDecl
  T8, Data from DeclDataDecl
  T9, Data from DeclDataDecl
  Left, Constructor from DeclDataDecl
  Right, Constructor from DeclDataDecl
  T10, Constructor from DeclDataDecl
  T11, Constructor from DeclDataDecl
  T3, Constructor from DeclDataDecl
  T5, Constructor from DeclDataDecl
  T6, Constructor from DeclDataDecl
  T7, Constructor from DeclDataDecl
  T8, Constructor from DeclDataDecl
  collection, Value from DeclDataDecl
  field1, Selector from DeclDataDecl
  field2, Selector from DeclDataDecl
